pkg/runner/run_context.go

```diff
@@ -782,37 +785,36 @@ func (rc *RunContext) localCheckoutPath() (string, bool) {
 	return "", false
 }
 
-func (rc *RunContext) handleCredentials(ctx context.Context) (string, string) {
-	username := rc.Config.Secrets["DOCKER_USERNAME"]
-	password := rc.Config.Secrets["DOCKER_PASSWORD"]
+func (rc *RunContext) handleCredentials() (username, password string, err error) {
+	// TODO: remove below 2 lines when we can release act with breaking changes
+	username = rc.Config.Secrets["DOCKER_USERNAME"]
+	password = rc.Config.Secrets["DOCKER_PASSWORD"]
 
 	container := rc.Run.Job().Container()
 	if container == nil {
-		return username, password
+		return
```

Comment:
I think `err` is never returned. But I see what you want to do, I can make it work.

Reply:
`err` at this point is `nil`, we want to escape `handleCredentials` when `container:` is not used and that should be an error